src/main/java/appeng/helpers/iface/PatternProviderLogicHost.java

```diff
@@ -46,6 +46,8 @@
 public interface PatternProviderLogicHost extends IConfigurableObject, IPriorityHost, PatternContainer {
     PatternProviderLogic getLogic();
 
+    PatternProviderLogic createLogic();
```

Comment:
It's what I went for as a factory method as discussed on Discord, but I wasn't sure if this was strictly required here or if it would work just fine as an internal class method to override all the same.